python/mlx/nn/losses.py

```diff
@@ -24,7 +35,42 @@ def cross_entropy(
     return _reduce(loss, reduction)
 
 
-def l1_loss(predictions: mx.array, targets: mx.array, reduction: str = "none") -> mx.array:
+def binary_cross_entropy(
+    inputs: mx.array, targets: mx.array, reduction: str = "none"
+) -> mx.array:
+    """
+    Computes the binary cross entropy loss between inputs and targets.
+
+    Args:
+        inputs (mx.array): The predicted inputs.
+        targets (mx.array): The target values.
+        reduction (str, optional): Specifies the reduction to apply to the output:
+          ``'none'`` | ``'mean'`` | ``'sum'``. Default: ``'none'``.
+
+    Returns:
+        mx.array: The computed binary cross entropy loss.
+    Examples:
+        >>> import mlx.core as mx
+        >>> import mlx.nn as nn
+        >>> inputs = mx.array([0.1, 0.2, 0.3, 0.4])
+        >>> targets = mx.array([0, 0, 1, 1])
+        >>> loss = nn.losses.binary_cross_entropy(inputs, targets)
+        >>> loss
+        array([0.612192])
+    """
+    loss = -targets * mx.log(inputs) - (1 - targets) * mx.log(1 - inputs)
+
+    return _reduce(loss, reduction)
+
+
+@_make_loss_module(binary_cross_entropy)
+class BCELoss(Module):
+    pass
+
```

Comment:
Cool! Could you add a test using the module as well, just to make sure the "module making" code works?

Reply:
ofc, on it!